sw/device/rom_exts/rom_ext_start.S

```diff
@@ -117,12 +119,115 @@ _rom_ext_interrupt_vector:
   .option norelax
 
 /**
- * Entry point after reset. This symbol is jumped to from the handler
- * for IRQ 32.
- *
- * At the moment, this should just fault, until we have a CRT we can use.
+ * Entry point. This symbol is jumped to from the handler the Mask ROM
+ * (execution handover to ROM_EXT).
  */
   .globl _rom_ext_start_boot
   .type _rom_ext_start_boot, @function
 _rom_ext_start_boot:
-  unimp
+
+  /**
+   * Disable Interrupts.
+   *
+   * We cannot disable exceptions, or Ibex's non-maskable interrupts (interrupt
+   * 31), so we still need to be careful.
+   */
+
+  // Clear `MIE` field of `mstatus` (disable interrupts globally). 
+  csrci mstatus, 0x8
+
+  // Clear all the machine-defined interrupts, `MEIE`, `MTIE`, and `MSIE` fields
+  // of `mie`.
+  li   t0, 0xFFFF0888
+  csrc mie, t0
+
+  la   sp, _stack_start
```

Comment:
I'm a little bit worried that this code is now duplicated three or four times in the repo. Could we file an issue to factor this out into a file(s) intended to be `#included` into crt0-type files? Only one of the copies (the one for flash binaries) is tested, and I don't want us to have a nasty surprise when one of the untested ones silently breaks.

Reply:
This seems pretty reasonable, I don't like the duplication especially if there's only the one test and it's not being run against everything.

Reply:
I think we should address this in the future. We know `boot_rom` and the existing `flash` stuff is likely going away, once we have enough Mask ROM etc in place to do bootstrapped builds. I realise your worries about this code (I have the same), but in the short term I think it's easier to review with duplication.

It is not my intention to update the `boot_rom_start.S` or the `flash_crt.S` if we can help it, at this point.

Reply:
I think it is just easier to move these things forward in smaller steps. The description of the PR says that this is an early stage, however @mcy this is a good suggestion that I also have thought about. I think we should capture it in an issue and have as a follow-up.

What do you guys think, is this a reasonable approach in your opinion?

I agree with @lenary , we don't have yet the complete vision of how Mask ROM and ROM_EXT start-up code should look like, so I think it is a bit premature to merge the common bits (just to avoid unnecessary churn).

Reply:
Added am item in #3875 . Resolving.